### PR TITLE
chore(npm): enforcing npm version

### PR DESCRIPTION
**Pull request by dominikfeininger**
not bulletproof yet, but happy to get suggestions ;)

### CODE REVIEW

File: check-versions.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const exec = require("child_process").exec
```
Comment (vobu) [inline check-versions.js:1]: const { exec, execSync } = require("child_process")

File: check-versions.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const npmVersion = parseFloat(stdout)
+
+    sReqNpmVersion = sReqNpmVersion.replace("<", "")
+    sReqNpmVersion = sReqNpmVersion.replace("=", "")
+    sReqNpmVersion = sReqNpmVersion.replace(">", "")
+    const fReqNpmVersion = parseFloat(sReqNpmVersion)
+
+    if (fReqNpmVersion > npmVersion) {
+        console.info(
+            `updating npm version from ${npmVersion} to the lasted version because required npm version in package.json is set to ${sReqNpmVersion}`
```
Comment (vobu) [inline check-versions.js:21]: typo: "latest"

**Review by vobu — CHANGES_REQUESTED** [review]
please also 
- check the code scan findings 
- the linter gh action output

**Comment by dominikfeininger** [comment]
I noticed that the `engine-strict` prevents the execution of the `check-versions` script if the npm version is too low. So I guess we need to decide which way we would like to go.

**Comment by vobu** [comment]
closing this as obsolete und not relevant any more.
docs point to npm >= 7 → every dev manages to comply ;)

**Pull request closed without merging**